### PR TITLE
[WIP] Add ini filters to_ini and from_ini

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY

This PR adds `to_ini` and `from_ini` filters.

I've had this sitting around for nearly a year, might as well start doing something with it.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
filters/ini

##### ANSIBLE VERSION

```
devel
```

##### ADDITIONAL INFORMATION
N/A

### CODE REVIEW

**Comment by mmckinst** [comment]
FWIW I think this would be a really nice feature since so many config files are just ini files. You could get rid of lots of templates and just use the `copy` module with the content parameter to write YAML out to an INI.

**Comment by p53** [comment]
+1

File: docs/docsite/rst/user_guide/playbooks_filters.rst [hunk 1]
```diff
@@ -62,6 +64,10 @@ for example::
        msg: '{{ item }}'
     loop: '{{ result.stdout | from_yaml_all | list }}'
 
+.. versionadded:: 2.8
+
+The ``to_ini`` and ``from_ini`` allow converting a dict to ini format, and ini format to a dict.
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_filters.rst:69]: The `to_ini` and `from_ini` filters fit nicely with the to/from JSON and to/from YAML filters. I'd reorganize the docs content a bit:

- put the `version_added` notes together - something like 
```
.. versionadded:: 2.7 (to_json, from_json, to_yaml, from_yaml)
.. versionadded:: 2.8 (to_ini, from_ini)
```
- remove the sentence above from this doc
- add the sentence above to the 2.8 porting guide
- add the sentence above to a changelog fragment

Comment (acozine) [reply docs/docsite/rst/user_guide/playbooks_filters.rst:69]: Hm, thinking it over, I'd put all the `versionadded` notes right under the header for `Filters For Formatting Data`.

**Comment by samccann** [comment]
Hey @sivel - any updates for this PR? Looks to be around for some time :-)

**Comment by sivel** [comment]
@samccann It was in WIP for a long time.  I recently decided to polish up the PR, but we have not decided if we actually want to merge this or not.

I've just put it back in WIP for now.

I plan to keep it around, and we may decided to add it at some point in the future.

**Review by jamescassell — APPROVED** [review]
shipit

Would love to see this merged!

**Comment by wainersm** [comment]
@sivel I'm working on a playbook that gets the user input formatted as multi-line key=value string. In order to get a key value I do something like this: 
```
{{ "key1=val1\nkey2=val2" | regex_search('^key1=.*', multiline=True) | replace('key1=','') }}
``'
That's the best approach I've found using current Ansible filters. Notice that I don't expect any ini section, otherwise the solution would be completely different (maybe write to a temporary file then read it with ini lookup?).

Your proposed `from_ini` simplifies a lot my code. Yet I believe it is a very common scenario so others would benefit from this merge. That said, what can I do to help you finish this work and hopefully get it merged?

**Comment by sivel** [comment]
We have decided to not merge this for now.  I have began work on migrating these filters to a collection that can be used to get access to them.  Once I get that collection published, I'll link back here and close this issue, but we won't see this PR merged.

**Comment by wainersm** [comment]
> We have decided to not merge this for now. I have began work on migrating these filters to a collection that can be used to get access to them. Once I get that collection published, I'll link back here and close this issue, but we won't see this PR merged.

@sivel thanks for the prompt update!

**Comment by sivel** [comment]
These filters have been published to https://galaxy.ansible.com/sivel/toiletwater

**Pull request closed without merging**